### PR TITLE
Refactor clients, fix retry on non-transient error codes (like 404)

### PR DESCRIPTION
**Pull request by NickolausDS**
A bit of refactoring to remove duplicated code between the online/gpu flows. The combine base client contains the fix for not doing retries on 404.

### CODE REVIEW

File: gladier_xpcs/flow_base.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            cfg = self.get_cfg(private=True)
+            cid = self.funcx_client.register_container(**container)
+            fxid = self.funcx_client.register_function(function, function.__doc__, container_uuid=cid)
+            cfg[self.section][fxid_name] = fxid
+            cfg[self.section][fxck_name] = self.get_funcx_function_checksum(function)
+            cfg.save()
+        else:
+            super().register_funcx_function(function)
+
+    def retry_backoff(self, method, *args, **kwargs):
```
Comment (ravescovi) [inline gladier_xpcs/flow_base.py:36]: I fell like this should go into gladier base.

**Review by ravescovi — COMMENTED** [review]
Since we are moving to have full `flow_base` files. Could we call it `flow_online_base` so we can have other flows using the same structure in the future?
I would suggest that the files then would be called
`flow_online_base.py`
`xpcs_client_online_gpu.py`
`xpcs_client_online.py`

File: gladier_xpcs/flow_base.py (hunk 1 above)
Comment (NickolausDS) [reply gladier_xpcs/flow_base.py:36]: Really, it won't be needed in a week or so. I asked the automate flow about automatic retries, and it's automatic in the next version. We just need to wait a few days for them to do the stable release.

**Comment by NickolausDS** [comment]
Let's rename the client classes in a different PR, this one is only meant to fix the 404 error.

**Comment by NickolausDS** [comment]
Why did you close this Raf? The automate update is a few days off, and this is a problem we're having now with the GPU runs.

**Comment by ravescovi** [comment]
Sorry. I understood we were going to wait for it.
Merge it then.

**Pull request merged**